decimal_test.go

```diff
@@ -2344,6 +2344,19 @@ func TestBinary(t *testing.T) {
 	}
 }
 
+func TestBadBinary(t *testing.T) {
```

Comment:
Also, could we update the test case name to something more verbose too? `BadBinary` is not telling much in my opinion. This test case explicitly checking if input data is long (5 bytes) enough.